src/debug_utils.cc

```diff
@@ -93,42 +93,112 @@ NativeSymbolDebuggingContext::New() {
+
+    if (SymFromAddr(hProcess, dwAddress, &dwDisplacement, pSymbol)) {
+      // SymFromAddr returned success
+      return NameAndDisplacement(pSymbol->Name, dwDisplacement);
+    } else {
+      // SymFromAddr failed
+      const DWORD error = GetLastError();
+#ifdef _DEBUG
+      fprintf(out_, "SymFromAddr returned error : %lu\n", error);
+#endif
+    }
+    // End MSDN code
 
-    info->MaxNameLen = MAX_SYM_NAME;
-    info->SizeOfStruct = sizeof(SYMBOL_INFO);
+    return NameAndDisplacement();
+  }
 
-    SymbolInfo ret;
-    const bool have_info = SymFromAddr(current_process_,
-                                       reinterpret_cast<DWORD64>(address),
-                                       nullptr,
-                                       info);
-    if (have_info && strlen(info->Name) == 0) {
-      if (UnDecorateSymbolName(info->Name,
-                               demangled,
-                               sizeof(demangled),
-                               UNDNAME_COMPLETE)) {
-        ret.name = demangled;
-      } else {
-        ret.name = info->Name;
-      }
+  SymbolInfo WrappedGetLine(DWORD64 dwAddress) const {
+    SymbolInfo sym{};
+    HANDLE hProcess = current_process_;
+
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-symbol-information-by-address
+    // Patches:
+    // * Use `fprintf(out, ` instead of `printf`
+    // * `sym ~= line` on success
+
+    // Patch: made into arg - DWORD64  dwAddress;
+    DWORD  dwDisplacement;
+    IMAGEHLP_LINE64 line;
+
+    SymSetOptions(SYMOPT_LOAD_LINES);
+
+    line.SizeOfStruct = sizeof(IMAGEHLP_LINE64);
+    // Patch: made into arg - dwAddress = 0x1000000;
+
+    if (SymGetLineFromAddr64(hProcess, dwAddress, &dwDisplacement, &line)) {
+      // SymGetLineFromAddr64 returned success
+      sym.filename = line.FileName;
+      sym.line = line.LineNumber;
+    } else {
+      // SymGetLineFromAddr64 failed
+      DWORD error = GetLastError();
+#ifdef _DEBUG
+      fprintf(out_, "SymGetLineFromAddr64 returned error : %lu\n", error);
+#endif
+    }
+    // End MSDN code
+
+    return sym;
+  }
+
+  // Fills the SymbolInfo::name of the io/out argument `sym`
+  std::string WrappedUnDecorateSymbolName(const char* name) const {
+    char szUndName[MAX_SYM_NAME];
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-undecorated-symbol-names
+    // Patches:
+    // * `ret.name = szUndName` instead of `printf`
+    if (UnDecorateSymbolName(name, szUndName, sizeof(szUndName),
+                             UNDNAME_COMPLETE)) {
+      // UnDecorateSymbolName returned success
+      return szUndName;
+    } else {
+      // UnDecorateSymbolName failed
+      const DWORD error = GetLastError();
+#ifdef _DEBUG
+      fprintf(out_, "UnDecorateSymbolName returned error %lu\n", error);
+#endif
     }
+    return nullptr;
+  }
 
+  SymbolInfo LookupSymbol(void* address) override {
+    auto dwAddress = reinterpret_cast<DWORD64>(address);
```

Comment:
`DWORD64 dw_address`

Reply:
Ack.
But the type stated explicitly on the rhs.